bean628/MyProject/app/src/main/java/com/practice/achitecture/myproject/MainActivity.kt

```diff
@@ -1,12 +1,109 @@
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.practice.achitecture.myproject.model.ResultOfSearchingModel
+import com.practice.achitecture.myproject.network.RetrofitClient
+import kotlinx.android.synthetic.main.activity_main.*
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+class MainActivity : AppCompatActivity(), View.OnClickListener {
 
-class MainActivity : AppCompatActivity() {
+    private var mSearchType: Int = 0    // 0: 영화, 1: 책, 2: 블로그, 3: 뉴스
+    private lateinit var mRetrofitClient: RetrofitClient
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+
+        btn_search.setOnClickListener(this)
+        btn_search_type_movie.setOnClickListener(this)
+        btn_search_type_book.setOnClickListener(this)
+        btn_search_type_blog.setOnClickListener(this)
+        btn_search_type_news.setOnClickListener(this)
+        input_search_sth.setOnEditorActionListener { v, actionId, event ->
+            when (actionId) {
+                EditorInfo.IME_ACTION_SEARCH -> {
+                    search(mSearchType)
+                    true
+                }
+                else -> false
+            }
+        }
+
+        mRetrofitClient = RetrofitClient()
+
+        rv_searched_list.layoutManager = LinearLayoutManager(this)
+    }
+
+    override fun onClick(v: View?) {
+
+        when (v?.id) {
+            R.id.btn_search -> search(mSearchType)
+            R.id.btn_search_type_movie ->
+                search(0)
+            R.id.btn_search_type_book ->
+                search(1)
+            R.id.btn_search_type_blog ->
+                search(2)
+            R.id.btn_search_type_news ->
+                search(3)
+        }
+
     }
+
+    private fun search(searchType: Int) {
+        val imm: InputMethodManager =
+            getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
+        imm.hideSoftInputFromWindow(input_search_sth.windowToken, 0)
+
+        val word: String = input_search_sth?.text.toString()
+        if (word.isEmpty()) {
+            Toast.makeText(this@MainActivity, "검색어를 입력해주세요.", Toast.LENGTH_SHORT).show()
+            return
+        }
+        mSearchType = searchType
+        val category = when (searchType) {
+            0 -> "movie"
+            1 -> "book"
+            2 -> "blog"
+            3 -> "news"
+            else -> "movie"
+        }
+
+        val call: Call<ResultOfSearchingModel> =
+            mRetrofitClient.retrofitInterface.searchSomething(category, word)
+
+        call.enqueue(object : Callback<ResultOfSearchingModel> {
+            override fun onResponse(
+                call: Call<ResultOfSearchingModel>,
+                response: Response<ResultOfSearchingModel>
+            ) {
+                if (response.code() == 200) {
```

Comment:
`isSuccessful` 을 사용하시면 더 좋을 것 같아요.
```suggestion
                if (response.isSuccessful) {
```

Reply:
수정했습니다

3445d76